web/src/components/MapView.tsx

```diff
@@ -28,19 +34,92 @@ const layerStyle: LayerProps = {
   paint: {}
 }
 
+const renderPopup = (coordinates: Position, description: string): ReactElement => {
+  return (
+    <StyledPopup
+      longitude={coordinates[0]}
+      latitude={coordinates[1]}
+      closeButton={false}
+      closeOnClick={false}
+      anchor='bottom'>
+      <div dangerouslySetInnerHTML={{ __html: description }} />
+    </StyledPopup>
+  )
+}
+
+const getPopUpDescription = (properties: GeoJsonProperties): string => {
+  return `<div><strong>${properties?.title}</strong></div>`
+}
+
+const getCursor = ({ isHovering, isDragging }: { isHovering: boolean; isDragging: boolean }): Property.Cursor => {
+  return isDragging ? 'grabbing' : isHovering ? 'pointer' : 'default'
+}
+
 interface MapViewProps {
-  featureCollection: FeatureCollection
+  featureCollection: FeatureCollection<Point, GeoJsonProperties>
 }
 
 const MapView: React.FunctionComponent<MapViewProps> = (props: MapViewProps): ReactElement => {
-  const [viewport, setViewport] = useState(defaultViewportConfig)
+  const [viewport, setViewport] = useState<MapViewViewport>(defaultViewportConfig)
+  const [showPopup, togglePopup] = React.useState<boolean>(false)
+  const [currentPoi, setCurrentPoi] = React.useState<Feature<Point, GeoJsonProperties> | null>(null)
+
   const { featureCollection } = props
+  const query = new URLSearchParams(useLocation().search)
+  const queryId = Number(query.get(mapParam))
+
+  const getCurrentPoi = useCallback(
+    (
+      features: FeatureCollection<Point, GeoJsonProperties>,
+      queryId: number
+    ): Feature<Point, GeoJsonProperties> | undefined => {
+      return featureCollection.features.find(feature => feature.properties?.id === queryId)
+    },
+    [featureCollection.features]
+  )
+
+  useEffect(() => {
+    const currentPoi: Feature<Point, GeoJsonProperties> | undefined = getCurrentPoi(featureCollection, queryId)
```

Comment:
I don't really see a reason to create a separate function for getting POIs here. As it would recteate the useEffect and the useCallback both when the features update and you also only use it once. See https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
yes you are right :) i mean i could use the fkt outside of the functional component and pass values but indeed its not reused anywhere (yet)